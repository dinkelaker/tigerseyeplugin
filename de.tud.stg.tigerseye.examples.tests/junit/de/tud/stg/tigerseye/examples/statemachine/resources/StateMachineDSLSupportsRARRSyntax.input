state "reseted" type "start" {
  entry "resetTimer"
  transitions {
  	"start" → "running" 
  	"switchOff" → "off"
  }
};

state "running" {
  entry "startTimer"
  perform "loopRunning" //do is a reserved keyword in Groovy
  transitions {
  "split" → "paused"
  "stop" → "stopped"
  }
};

state "paused" {
  entry "pauseTimer"
  transitions {
  	"unsplit" → "running"
  	"stop" → "stopped"
  }
};

state "stopped" {
  entry "stopTimer"
  transitions {
  "reset" → "stopped"
  "switchOff" → "off"
  }
};

state "off" {
  exit "switchOff"
  transitions {
	→ "end"
  }
};
