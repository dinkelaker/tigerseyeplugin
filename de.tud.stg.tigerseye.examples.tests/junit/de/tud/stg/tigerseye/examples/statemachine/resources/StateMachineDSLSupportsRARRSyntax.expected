
		state_name_type_closure("reseted","start") {
		  entry "resetTimer"
		  transitions {
			when_event_enter("start","running")
			when_event_enter("switchOff","off")
		  }
		};
	
		state_name_closure("running") {
		  entry "startTimer"
		  perform: "loopRunning" //do is a reserved keyword in Groovy
		  transitions {
			when_event_enter("split","paused")
			when_event_enter("stop","stopped")
		  }
		};
	
		state_name_closure("paused") {
		  entry "pauseTimer"
		  transitions {
			when_event_enter("unsplit","running")
			when_event_enter("stop","stopped")
		  }
		};
	
		state_name_closure("stopped") {
		  entry "stopTimer"
		  transitions {
			when_event_enter("reset","stopped")
			when_event_enter("switchOff","off")
		  }
		};
	
		state_name_closure("off") {
		  exit "switchOff"
		  transitions {
			when_enter("end")
		  }
		};